### ISSUE TITLE
Filament MultiTenancy Support

### ISSUE THREAD

**Issue by craigvc** [visualbuilder/email-templates#6]
This is a great package but are you going to support multi-tenant filament projects?

**Comment by cannycookie** [comment]
What would you like to do with multi tenants and email templates? I haven't had a use case come up yet for us.

**Comment by craigvc** [comment]
well we have travel agencies as tenant, so we want each agency to be able to customise their own templates for mails,

**Comment by cannycookie** [comment]
OK, Should be easy enough but presumably also need to handle system emails that don't belong to a tenant.

**Comment by craigvc** [comment]
Hi no just user/agency mailsSent from my iPhoneOn 19 Apr 2024, at 14:31, Lee ***@***.***> wrote:﻿
OK, Should be easy enough but presumably also need to handle system emails that don't belong to a tenant.

**Comment by willcastillo** [comment]
Just came to suggest this as well so plus one.

Thanks for this plugin @cannycookie !

**Comment by willcastillo** [comment]
I'm not sure I understand the last message from Craig but for me, yes, it'd be great if the same plugin can be use to handle both tenants and also system/application's emails

**Comment by FrankWiebe** [comment]
I'm looking for this as well.

**Comment by Cyrille37** [comment]
Hi
Thanks for this great plugin 💌 . Like Filament 3 has bringing [multi-tenancy](https://filamentphp.com/docs/3.x/panels/tenancy), plugins should propose this compatibility 🥰  .

**Comment by systemEDOM** [comment]
any idea how to resolve the issue in filament with multi tenancy? Package is totally failing

**Comment by cannycookie** [comment]
> any idea how to resolve the issue in filament with multi tenancy? Package is totally failing

In principle just add a tenant_id to the email templates table and setup a global filter where tenant_id = filament tenant id.

Theres a little more to it than that as you'll need to set permissions on who can edit the tenant id so there needs to be an admin that can view & edit, while tenants should not be able to see it or change it.  

I'm using the the package on a couple of multitenancy sites but tenants don't have permission to edit the templates, so it's not broken. Its' just not setup to allow tenants ability to edit their own templates.

I do intend to add mulitenancy but much of the code will be dependent on how you're implementing permissions for tenants in your code.

**Comment by tndejong** [comment]
Maybe first just only make multi tenancy check 'protected static bool $isScopedToTenant = false;' on the resources possible to set through the config. So the templates are not by default scoped to your multi tenant query.

**Comment by lochinvarwest** [comment]
Agreed @tndejong.  Then I'd also suggest config fields "isMultiTenant"  (obviously) and  "tenancyModel" (or the like).
Then its a matter of including the relations on the models (template belongsTo tenant and tenant hasMany templates) (and adjusting the migrations of course).
And then Filament's multi-tenancy should handle most of the rest...  (or at least, on first attempt on my side... this seems to be is working).
